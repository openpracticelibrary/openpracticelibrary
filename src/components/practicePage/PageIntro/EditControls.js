import React from "react";
import { useTheme } from "@material-ui/core/styles";
import { CloseIcon, Edit, SaveOutlinedIcon } from "@material-ui/icons";
import { IconButton, Box } from "@material-ui/core";

import OplTooltip from "../../shared/components/OplTooltip";

const StartEditingButton = (props) => (
  <Box ml={-6}>
    <OplTooltip title="Edit Practice" placement="top">
      <IconButton data-testid="editButton" onClick={props.handleEdit}>
        <Edit />
      </IconButton>
    </OplTooltip>
  </Box>
);

const EditingButtons = (props) => {
  const theme = useTheme();

  return (
    <Box
      display="flex"
      flexDirection="column"
      bgcolor="white"
      borderRadius={20}
      boxShadow="0 1px 4px 0 rgba(16,16,16,0.29)"
      ml={-6}
    >
      <IconButton>
        <Edit color={theme.palette.primary.main} />
      </IconButton>
      <OplTooltip title="Save Edits" placement="left">
        <IconButton
          data-testid="saveEditButton"
          onClick={props.handleSaveEdits}
        >
          <SaveOutlinedIcon />
        </IconButton>
      </OplTooltip>
      <OplTooltip title="Cancel Edits" placement="left">
        <IconButton data-testid="closeEditButton" onClick={props.handleEdit}>
          <CloseIcon />
        </IconButton>
      </OplTooltip>
    </Box>
  );
};

export { StartEditingButton, EditingButtons };
