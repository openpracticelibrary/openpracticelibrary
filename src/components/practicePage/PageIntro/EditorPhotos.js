import React from "react";
import { withTheme } from "@material-ui/core/styles";
import { AvatarGroup } from "@material-ui/lab";
import { Person } from "@material-ui/icons";
import OplAvatar from "../../shared/components/OplAvatar";

const EditorPhotos = (props) => (
  <AvatarGroup max={3}>
    {props.editors.map((editor, i) => (
      <OplAvatar
        key={i}
        src={`https://github.com/${editor.github}.png`}
        height={props.theme.spacing(4)}
        width={props.theme.spacing(4)}
        bgcolor="transparent"
      >
        <Person />
      </OplAvatar>
    ))}
  </AvatarGroup>
);

export default withTheme(EditorPhotos);
