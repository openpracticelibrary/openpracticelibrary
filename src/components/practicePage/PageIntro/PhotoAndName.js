import React from "react";
import { withTheme } from "@material-ui/core/styles/index";
import { Person } from "@material-ui/icons";
import { Box } from "@material-ui/core";

import OplTypography from "../../shared/components/OplTypography";
import OplAvatar from "../../shared/components/OplAvatar";

const PhotoAndName = (props) => (
  <Box
    display="flex"
    flexDirection="row"
    justifyContent="flex-start"
    alignItems="flex-start"
    pr={3}
    minWidth="fit-content"
  >
    <OplAvatar
      src={props.avatar}
      width={props.theme.spacing(4)}
      height={props.theme.spacing(4)}
      bgcolor="transparent"
    >
      <Person />
    </OplAvatar>
    <OplTypography variant="body1" pl={1} data-testid={"authorName"}>
      {props.authorName}
    </OplTypography>
  </Box>
);

export default withTheme(PhotoAndName);
