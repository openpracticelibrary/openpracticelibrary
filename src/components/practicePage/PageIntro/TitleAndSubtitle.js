import React from "react";
import { Typography, Box, Grid, Chip } from "@material-ui/core";
import { LocalOffer } from "@material-ui/icons";

const TitleAndSubtitle = ({ subtitle, tags, title }) => (
  <Box display="flex" flexDirection="column" marginTop={3}>
    <Typography variant="h2" gutterBottom data-testid="title">
      {title}
    </Typography>
    <Box marginBottom={3}>
      <Grid container spacing={1}>
        {tags &&
          tags.map((t, i) => (
            <Grid item>
              <Chip icon={<LocalOffer />} label={`${t}`} key={i} size="small" data-testid={`tagLabel${t}`}/>
            </Grid>
          ))}
      </Grid>
    </Box>
    <Typography variant="h5" data-testid={"subtitle"}>
      {subtitle}
    </Typography>
  </Box>
);

export default TitleAndSubtitle;
