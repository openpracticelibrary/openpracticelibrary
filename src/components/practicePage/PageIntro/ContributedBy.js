import React from "react";
import moment from "moment";
import PhotoAndName from "./PhotoAndName";
import EditorPhotos from "./EditorPhotos";
import { Typography, Grid, Box } from '@material-ui/core';

export default function ContributedBy(props) {
  console.log(props);
  const formatDate = (date) => {
    return moment(date).format("MMMM D, YYYY");
  };
  let contributors = [];
  let editors = [];

  if (props.authors && props.authors.length > 2) {
    contributors = props.authors.slice(0, 2);
    editors = props.authors.slice(2, props.authors.length);
  } else {
    contributors = props.authors;
  }

  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="flex-start"
      alignItems="flex-start"
      my={2}
    >
      <Grid
        container
        direction="row"
        justify="flex-start"
        alignItems="flex-end"
      >
        <Box mr={2}>
          <Typography variant="overline">Contributed by</Typography>
          <Grid item>
            <Box display="flex" flexDirection="row">
              {contributors && contributors.map((author, i) => (
                <PhotoAndName
                  key={i}
                  authorName={author.title}
                  avatar={`https://github.com/${author.github}.png`}
                />
              ))}
            </Box>
          </Grid>
        </Box>
        { (editors && editors.length > 0) &&
        <Box mr={3}>
          <Typography variant="overline">Edited by</Typography>
          <Grid item>
            <EditorPhotos editors={editors} />
          </Grid>
        </Box>
        }
        <Grid item>
          <Typography variant="overline" data-testid={"dates"}>
                Published {formatDate(props.createdAt)} | Last edited{" "}
            {formatDate(props.updatedAt)}
          </Typography>
        </Grid>
      </Grid>
    </Box>
  );
}
