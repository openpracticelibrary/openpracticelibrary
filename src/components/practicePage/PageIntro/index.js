import React from "react";
import TitleAndSubtitle from "./TitleAndSubtitle";
import ContributedBy from "./ContributedBy";
import Collection from "../../shared/Collection";
import { Container, Typography } from "@material-ui/core";

const PageIntro = ({
  title,
  tags,
  subtitle,
  editing,
  titleEdit,
  subtitleEdit,
  authors,
  createdAt,
  updatedAt,
  practiceId,
  upvotes,
  imgCount,
  questions,
}) => (
  <Container maxWidth="md">
    <TitleAndSubtitle
      title={title}
      tags={tags}
      subtitle={subtitle}
      editing={editing}
      titleEdit={titleEdit}
      subtitleEdit={subtitleEdit}
    />
    <ContributedBy
      authors={authors}
      createdAt={createdAt}
      updatedAt={updatedAt}
    />

    <Collection
      spacing={2}
      practiceId={practiceId}
      upvotes={upvotes}
      imgCount={imgCount}
      questions={questions}
      alignment={2}
    >
      <Typography variant="overline">Collection</Typography>
    </Collection>
  </Container>
);

export default PageIntro;
