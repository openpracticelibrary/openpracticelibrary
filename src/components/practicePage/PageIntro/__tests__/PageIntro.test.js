import React from "react";
import {cleanup, render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {MockedProvider} from "@apollo/react-testing";
import renderer from 'react-test-renderer';
import PageIntro from '../index';

afterEach(cleanup);

const date = new Date(2020,6,8);

const mockProps = {
  title: "This is a test title",
  subtitle: "A subtitle for your title",
  slug: "event-storming",
  authors: [
    {
      firstName: "Tim",
      id: "5ec55d3bd36579001355faee",
      lastName: "Beattie",
      mediaLink: "https://github.com/tdbeattie",
    },
    {
      firstName: "Matt",
      id: "5ec55d09d36579001355faed",
      lastName: "Takane",
      mediaLink: "https://github.com/mtakane",
    }
  ],
  createdAt: date,
  updatedAt: date,
  practiceId: "aFak3ID4y0U",
  upvotes: 42,
  imgCount: 3,
  questions: 5,
};

it("renders Page Introduction without crashing", () => {
  render(
    <MockedProvider mocks={[]}>
      <PageIntro {...mockProps} />
    </MockedProvider>
  );
});

it("matches the snapshot", () => {
  const tree = renderer.create(
    <MockedProvider mocks={[]}>
      <PageIntro {...mockProps} />
    </MockedProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
