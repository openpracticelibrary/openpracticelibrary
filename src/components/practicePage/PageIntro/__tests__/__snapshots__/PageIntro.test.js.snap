// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="MuiGrid-root sc-AxiKw jHfFLS MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
  my={5}
>
  <div
    className="MuiGrid-root sc-AxiKw bOTXee MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-254"
    >
      <h3
        className="MuiTypography-root sc-AxjAm bOzOQA MuiTypography-h3"
        data-testid="title"
        py={1}
      >
        This is a test title
      </h3>
      <div
        className="MuiBox-root MuiBox-root-285"
      />
      <h6
        className="MuiTypography-root sc-AxjAm bzxMlJ MuiTypography-subtitle1"
        data-testid="subtitle"
        pt={2}
      >
        A subtitle for your title
      </h6>
    </div>
  </div>
  <div
    className="MuiGrid-root sc-AxiKw bOTXee MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-286"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end"
      >
        <div
          className="MuiBox-root MuiBox-root-287"
        >
          <span
            className="MuiTypography-root MuiTypography-overline"
          >
            Contributed by
          </span>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiBox-root MuiBox-root-288"
            >
              <div
                className="MuiBox-root MuiBox-root-289"
              >
                <div
                  bgcolor="transparent"
                  className="MuiAvatar-root MuiAvatar-circle sc-AxirZ eyyuTD MuiAvatar-colorDefault"
                  height={24}
                  width={24}
                >
                  <svg
                    height="24"
                    viewBox="0 0 25 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.696 0C6.084 0 .704 5.383.704 12s5.38 12 11.992 12c6.612 0 11.992-5.383 11.992-12S19.308 0 12.696 0zm2.058 17.213c1.863-1.444 3.17-4.571 3.17-6.771 0-2.946-2.341-5.334-5.228-5.334-2.887 0-5.227 2.388-5.227 5.334 0 2.2 1.306 5.327 3.169 6.771-2.584.34-4.647 1.308-5.49 2.551-1.874-1.948-3.033-4.616-3.033-7.556 0-5.953 4.747-10.796 10.581-10.796s10.58 4.843 10.58 10.796c0 2.94-1.158 5.608-3.032 7.557-.843-1.244-2.905-2.211-5.49-2.552z"
                      fill="#A0A0A0"
                      fillRule="nonzero"
                    />
                  </svg>
                </div>
                <p
                  className="MuiTypography-root sc-AxjAm hTFVJK MuiTypography-body1"
                  data-testid="authorName"
                  pl={1}
                >
                  Tim Beattie
                </p>
              </div>
              <div
                className="MuiBox-root MuiBox-root-297"
              >
                <div
                  bgcolor="transparent"
                  className="MuiAvatar-root MuiAvatar-circle sc-AxirZ eyyuTD MuiAvatar-colorDefault"
                  height={24}
                  width={24}
                >
                  <svg
                    height="24"
                    viewBox="0 0 25 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.696 0C6.084 0 .704 5.383.704 12s5.38 12 11.992 12c6.612 0 11.992-5.383 11.992-12S19.308 0 12.696 0zm2.058 17.213c1.863-1.444 3.17-4.571 3.17-6.771 0-2.946-2.341-5.334-5.228-5.334-2.887 0-5.227 2.388-5.227 5.334 0 2.2 1.306 5.327 3.169 6.771-2.584.34-4.647 1.308-5.49 2.551-1.874-1.948-3.033-4.616-3.033-7.556 0-5.953 4.747-10.796 10.581-10.796s10.58 4.843 10.58 10.796c0 2.94-1.158 5.608-3.032 7.557-.843-1.244-2.905-2.211-5.49-2.552z"
                      fill="#A0A0A0"
                      fillRule="nonzero"
                    />
                  </svg>
                </div>
                <p
                  className="MuiTypography-root sc-AxjAm hTFVJK MuiTypography-body1"
                  data-testid="authorName"
                  pl={1}
                >
                  Matt Takane
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <span
            className="MuiTypography-root MuiTypography-overline"
            data-testid="dates"
          >
            Published 
            July 8, 2020
             | Last edited
             
            July 8, 2020
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root sc-AxiKw bOTXee MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-298"
    >
      <span
        className="MuiTypography-root sc-AxjAm buxdJi MuiTypography-overline"
        pr={2}
      >
        Collection
      </span>
      <h6
        className="MuiTypography-root sc-AxjAm geEeHi MuiTypography-subtitle1"
        data-testid="heartIcon"
        fontWeight="bold"
        onClick={[Function]}
      >
        <svg
          data-testid="heartSvg"
          height="17"
          viewBox="0 0 20 17"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.996 5.815C18.908 3.158 16.835 1 14.17 1c-1.778 0-3.33.958-4.169 2.384C9.162 1.958 7.609 1 5.83 1 3.166 1 1.092 3.158 1.005 5.815.804 11.841 10 16.187 10 16.187s9.196-4.346 8.996-10.372z"
            fill="#303030"
            stroke="#9FA4A3"
          />
        </svg>
         
        42
         
      </h6>
      <h6
        className="MuiTypography-root sc-AxjAm loJYex MuiTypography-subtitle1"
        data-testid="cameraIcon"
        fontWeight="bold"
        px={2}
      >
        <svg
          height="16"
          viewBox="0 0 22 16"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#303030"
          >
            <path
              d="M19.565 2.754H16.45L14.493.724C14.058.29 13.478 0 12.899 0H8.84c-.58 0-1.16.29-1.595.725L5.29 2.754H2.174C.942 2.754 0 3.696 0 4.928v8.84c0 1.232.942 2.174 2.174 2.174h17.391c1.232 0 2.174-.942 2.174-2.174v-8.84c0-1.16-.942-2.174-2.174-2.174zm-8.695 10.87c-2.682 0-4.856-2.175-4.856-4.856s2.174-4.855 4.856-4.855c2.68 0 4.855 2.174 4.855 4.855 0 2.681-2.174 4.855-4.855 4.855z"
            />
            <circle
              cx="10.87"
              cy="8.768"
              r="3.406"
            />
          </g>
        </svg>
         
        3
         
      </h6>
      <h6
        className="MuiTypography-root sc-AxjAm geEeHi MuiTypography-subtitle1"
        data-testid="questionIcon"
        fontWeight="bold"
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 18 15"
          width="18px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            6E24265F-DCD4-4060-AD24-1D9A528732F8
          </title>
          <g
            fill="none"
            fillRule="evenodd"
            id="Practice-Page---Product"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="#303030"
              fillRule="nonzero"
              id="OPL_Practice-Pg--All-Content"
              transform="translate(-438.000000, -310.000000)"
            >
              <g
                id="Practice-Intro---Wide"
              >
                <g
                  id="collection"
                >
                  <g
                    transform="translate(265.815406, 304.000000)"
                  >
                    <g
                      id="icons"
                      transform="translate(87.938931, 0.000000)"
                    >
                      <g
                        id="noun_Question_2965389"
                        transform="translate(94.434421, 13.500000) scale(-1, 1) translate(-94.434421, -13.500000) translate(86.939625, 6.000000)"
                      >
                        <path
                          d="M11.5828654,0 C12.4863851,0 13.352899,0.359171047 13.9917839,0.99849961 C14.6306688,1.63782817 14.9895906,2.50494378 14.9895906,3.40909091 L14.9895906,8.86363636 C14.9895906,10.7464253 13.4643478,12.2727273 11.5828654,12.2727273 L8.38054381,12.2727273 L4.46280992,14.8840909 C4.35195049,14.9586494 4.22163656,14.9989563 4.08807015,15 C3.97623357,14.9994408 3.86625402,14.9713414 3.76783799,14.9181818 C3.54616431,14.8000648 3.40739525,14.5694957 3.40672513,14.3181818 L3.40672513,12.2727273 C2.50320545,12.2727273 1.63669158,11.9135562 0.997806688,11.2742277 C0.358921795,10.6348991 -9.08863547e-13,9.76778349 -9.08863547e-13,8.86363636 L-9.08863547e-13,3.40909091 C-9.08863547e-13,1.52630199 1.52524279,0 3.40672513,0 L11.5828654,0 Z M7.40944938,8.57111111 C6.85427936,8.57111111 6.41014335,9.01555556 6.41014335,9.57111111 C6.41014335,10.1266667 6.85427936,10.5711111 7.40944938,10.5711111 C7.96461941,10.5711111 8.40875542,10.1266667 8.40875542,9.57111111 C8.40875542,9.01555556 7.96461941,8.57111111 7.40944938,8.57111111 Z M7.3720379,1.57111111 C6.92388781,1.57111111 6.53005895,1.64533792 6.1633907,1.78142039 C5.81030275,1.91750286 5.49795572,2.10306987 5.23992991,2.35049255 C4.98190411,2.59791523 4.77819952,2.88245132 4.64239647,3.21647194 C4.49301311,3.55049255 4.42511158,3.92162658 4.41153127,4.31750286 L4.41153127,4.31750286 L4.41924489,4.36035647 C4.4392351,4.4149874 4.49572917,4.45853379 4.56091463,4.47832761 C4.56091463,4.47832761 4.88684197,4.57729668 5.30783144,4.57729668 C5.72882092,4.57729668 6.05474825,4.47832761 6.05474825,4.47832761 C6.13623009,4.45358534 6.20413161,4.39172967 6.20413161,4.31750286 C6.20413161,4.31750286 6.21771192,3.93399771 6.27203314,3.76080183 C6.32635436,3.58760596 6.4078362,3.42678121 6.50289834,3.30306987 C6.61154078,3.17935853 6.73376353,3.06801833 6.8967272,2.99379152 C7.05969087,2.91956472 7.24981515,2.88245132 7.46710004,2.88245132 C7.79302737,2.88245132 8.05105318,2.96904926 8.22759715,3.129874 C8.40414112,3.29069874 8.49920326,3.55049255 8.49920326,3.88451317 C8.49920326,4.08245132 8.45846235,4.24327606 8.37698051,4.37935853 C8.28191837,4.51544101 8.17327593,4.63915235 8.03747287,4.75049255 C7.90166982,4.86183276 7.75228645,4.97317297 7.58932279,5.08451317 C7.42635912,5.19585338 7.27697576,5.33193585 7.1275924,5.48038946 C6.97820903,5.62884307 6.85598628,5.81441008 6.74734384,6.02471936 C6.63870139,6.24739977 6.57079987,6.50719359 6.54363925,6.82884307 L6.54363925,6.82884307 L6.54363925,7.32368843 C6.54363925,7.39791523 6.61154078,7.44739977 6.67944231,7.47214204 C6.67944231,7.47214204 6.99178934,7.57111111 7.3720379,7.57111111 C7.76586676,7.57111111 8.06463348,7.47214204 8.06463348,7.47214204 C8.14611532,7.44739977 8.20043654,7.39791523 8.20043654,7.32368843 L8.20043654,7.32368843 L8.20043654,6.90306987 C8.22759715,6.68038946 8.30907898,6.49482245 8.43130174,6.34636884 C8.55352449,6.19791523 8.70290785,6.06183276 8.86587151,5.95049255 C9.02883518,5.83915235 9.20537916,5.71544101 9.38192313,5.6041008 C9.57204741,5.48038946 9.73501107,5.34430699 9.88439444,5.18348225 C10.0337778,5.01028637 10.1695809,4.82471936 10.264643,4.58966781 C10.3732854,4.3669874 10.4276067,4.07008018 10.400446,3.69894616 C10.400446,3.48863688 10.3461248,3.25358534 10.2510627,3.00616266 C10.1560005,2.75873998 9.97945658,2.52368843 9.74859138,2.31337915 C9.50414588,2.10306987 9.19179885,1.929874 8.81155029,1.78142039 C8.43130174,1.64533792 7.94241073,1.57111111 7.3720379,1.57111111 Z"
                          id="Shape"
                          transform="translate(7.494795, 7.500000) scale(-1, 1) translate(-7.494795, -7.500000) "
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
         
        5
         
      </h6>
    </div>
  </div>
</div>
`;
